packages/hub/tasks/notify-customer-payment.ts

```diff
@@ -0,0 +1,34 @@
+import { inject } from '@cardstack/di';
+import CardpaySDKService from '../services/cardpay-sdk';
+import Web3HttpService from '../services/web3-http';
+import WorkerClient from '../services/worker-client';
+
+interface CustomerPaymentSubscriptionEvent {
+  returnValues: {
+    merchantSafe: string;
+    spendAmount: string;
+  };
+}
+
+export default class NotifyCustomerPayment {
+  cardpay: CardpaySDKService = inject('cardpay');
+  web3: Web3HttpService = inject('web3-http', { as: 'web3' });
+  workerClient: WorkerClient = inject('worker-client', { as: 'workerClient' });
+
+  async perform(payload: CustomerPaymentSubscriptionEvent) {
+    let Safes = await this.cardpay.getSDK('Safes', this.web3.getInstance());
```

Comment:
you'll need to wait for the subgraph to index the event. use `waitForSubgraphIndex` with the txn hash from the event